### PR TITLE
Complete the Environment class documentation

### LINKED ISSUE #34846: Unexpected "shadows" with SS reflections
**Godot version:**

Godot Engine v3.1.2.stable.custom_build
OpenGL ES 3.0 Renderer: GeForce GTX 970/PCIe/SSE2

**OS/device including version:**

Linux 5.4.7-arch1-1

**Issue description:**

A [reddit commenter](https://www.reddit.com/r/godot/comments/ek4xp7/sharing_progress_on_my_vehiclestrategy_game_to/fdb91lr?utm_source=share&utm_medium=web2x) pointed out some unusual visual artifacts in my game with SS reflections. It almost appears that a mesh is casting shadows over the reflections, but there's no light source behind the mesh that would be casting those shadows.

Possibly related to https://github.com/godotengine/godot/issues/9017, https://github.com/godotengine/godot/issues/15527, https://github.com/godotengine/godot/issues/17995

**Steps to reproduce:**

1. Create a plane with a `SpatialMaterial` that has `roughness=0`, `metallic=1`
2. Create a cube with a `SpatialMaterial` that has `emission=4`
3. In the environment, set `Background.mode=CustomColor=black`, and enable `SS Reflections`
4. The cube should reflect off the plane
5. Add another MeshInstance and move it around between the cube and the camera
6. Note that it blocks parts of the reflection, even if it is not between the cube and the plane

**Minimal reproduction project:**
[example.zip](https://github.com/godotengine/godot/files/4024111/example.zip)

^^^ note that the sphere is well above the cube in this picture, but it is taking a "chunk" out of the reflection.

### PR DESCRIPTION
**Pull request by Calinou**
This closes #34846.

### CODE REVIEW

**Comment by akien-mga** [comment]
Thanks!

**Review by clayjohn — COMMENTED** [review]
I started a review before this was merged. Posting for future reference

File: doc/classes/Environment.xml [hunk 1]
```diff
@@ -165,131 +168,151 @@
...
+			The intensity of the fog light transmittance effect. Amount of light that the fog transmits.
 		</member>
 		<member name="fog_transmit_enabled" type="bool" setter="set_fog_transmit_enabled" getter="is_fog_transmit_enabled" default="false">
-			Enables fog's light transmission. If enabled, lets reflections light to be transmitted by the fog.
+			Enables fog's light transmission effect. If [code]true[/code], light will be more visible in the fog to simulate light scattering as in real life.
 		</member>
 		<member name="glow_bicubic_upscale" type="bool" setter="set_glow_bicubic_upscale" getter="is_glow_bicubic_upscale_enabled" default="false">
-			Smooths out blockiness created by sampling higher levels.
+			Smooths out the blockiness created by sampling higher levels, at the cost of performance.
+			[b]Note:[/b] Only available when using the GLES3 renderer.
```
[the remark this answers is not in the document]
Comment (Calinou) [reply doc/classes/Environment.xml]: Out of curiosity, is that extension mandated by OpenGL ES 3.0?

[the remark this answers is not in the document]
Comment (clayjohn) [reply doc/classes/Environment.xml]: Not sure if the extension appears, but the content of it is part of the core GLES3 spec yes. I.e. bit shifting is a part of the GLES3 spec, but supported on many GLES2 devices due to the extension.

**Pull request merged**